test/integration/other/entities.js

```diff
@@ -0,0 +1,325 @@
+    it('should gracefully handle processing a non-entity submission', testService(async (service, { Entities, Submissions, one }) => {
+      await service.login('alice', (asAlice) =>
+        asAlice.post('/v1/projects/1/forms/simple/submissions')
+          .send(testData.instances.simple.one)
+          .set('Content-Type', 'application/xml')
+          .expect(200));
+
+      // Look up the submission to be able to get the corresponding form def
+      const subDef = await Submissions.getCurrentDefByIds(1, 'simple', 'one', false).then((s) => s.get());
+
+      // Processing a submission without an entity should not yield
+      // any errors but it should also not create any entities.
+      const entity = await Entities.processSubmissionDef(subDef.id);
+      should.not.exist(entity);
+      const { count } = await one(sql`select count(*) from entities`);
+      count.should.equal(0);
+    }));
+
+    it('should not make entity from non-entity approved submission', testService(async (service, container) => {
+      // When *any* submission is approved, the entity worker will look at it
+      // to determine if it contains an entity to be created.
+      await service.login('alice', (asAlice) =>
+        asAlice.post('/v1/projects/1/forms/simple/submissions')
+          .send(testData.instances.simple.one)
+          .set('Content-Type', 'application/xml')
+          .expect(200)
+          .then(() => asAlice.patch('/v1/projects/1/forms/simple/submissions/one')
+            .send({ reviewState: 'approved' })
+            .expect(200)));
+
+      await exhaust(container);
+
+      const { count } = await container.one(sql`select count(*) from entities`);
+      count.should.equal(0);
+    }));
+  });
+
+  it('should create an entity from a submission', testService(async (service, { Entities, Submissions, one }) => {
+    await service.login('alice', (asAlice) =>
+      asAlice.post('/v1/projects/1/forms?publish=true')
+        .send(testData.forms.simpleEntity)
+        .set('Content-Type', 'application/xml')
+        .expect(200)
+        .then(() => asAlice.post('/v1/projects/1/forms/simpleEntity/submissions')
+          .send(testData.instances.simpleEntity.one)
+          .set('Content-Type', 'application/xml')
+          .expect(200)));
+
+    // Look up the submission to be able to get the corresponding form def
+    const subDef = await Submissions.getCurrentDefByIds(1, 'simpleEntity', 'one', false).then((s) => s.get());
+
+    const entity = await Entities.processSubmissionDef(subDef.id);
+    entity.uuid.should.equal('12345678-1234-4123-8234-123456789abc');
+    entity.label.should.equal('Alice (88)');
+    entity.createdBy.should.equal(5); // submitter ID of Alice
+    entity.def.data.should.eql({ first_name: 'Alice', age: '88' });
+
+    const { count } = await one(sql`select count(*) from entities`);
+    count.should.equal(1);
+  }));
+
+  it('should log entity creation in audit log', testService(async (service, container) => {
```

Comment:
i like this test but i can be in a different file - maybe the entity worker test file.